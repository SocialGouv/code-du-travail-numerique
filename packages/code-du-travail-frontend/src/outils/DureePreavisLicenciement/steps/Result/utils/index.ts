export * from "./getResult";
export * from "./getSituations";
export * from "./getRefs";
export * from "./buildRecap";
export * from "./types";
