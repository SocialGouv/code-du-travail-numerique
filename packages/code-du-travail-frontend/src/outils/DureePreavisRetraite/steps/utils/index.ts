import { supportedCcn } from "@socialgouv/modeles-social";

import { AgreementSupportInfo } from "../../../common/Agreement/types";

export const getSupportedCC = (): AgreementSupportInfo[] =>
  supportedCcn.map((item) => ({
    fullySupported: item.preavisRetraite,
    idcc: item.idcc,
  }));
