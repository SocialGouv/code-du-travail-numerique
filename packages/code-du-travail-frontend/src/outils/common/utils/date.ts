import { parse as rootParse, parseISO } from "date-fns";

export const parse = (str: string) => {
  if (parseISO(str).toString() === "Invalid Date") {
    return rootParse(str, "dd/MM/yyyy", new Date());
  }
  return parseISO(str);
};

export default parse;
