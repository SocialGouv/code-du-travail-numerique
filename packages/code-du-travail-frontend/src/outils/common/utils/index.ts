export { default as parse } from "./date";
export * from "./math";
export { default as printResult } from "./printResult";
