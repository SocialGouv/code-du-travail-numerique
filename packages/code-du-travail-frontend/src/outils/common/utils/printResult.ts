import { MatomoBaseEvent, MatomoSimulatorEvent } from "../../../lib";
import { push as matopush } from "@socialgouv/matomo-next";

export default function printResult(simulatorTitle: string): void {
  matopush([
    MatomoBaseEvent.TRACK_EVENT,
    MatomoBaseEvent.OUTIL,
    MatomoSimulatorEvent.CLICK_PRINT,
    simulatorTitle,
  ]);
  const detailsTags = document.getElementsByTagName("details");
  for (let i = 0; i < detailsTags.length; i++) {
    detailsTags[i].setAttribute("open", "");
  }
  window.print();
}
