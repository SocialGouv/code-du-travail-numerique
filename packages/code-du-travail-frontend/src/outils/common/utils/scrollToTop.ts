export default function scrollToTop(): void {
  window.scrollTo(0, 0);
  window.parent?.postMessage({ kind: "scroll-to-top" }, "*");
}
