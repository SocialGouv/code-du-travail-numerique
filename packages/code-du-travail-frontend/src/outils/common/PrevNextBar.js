import { Button, icons, theme } from "@socialgouv/cdtn-ui";
import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

import printResult from "./printResult";

function PrevNextBar({
  onPrev,
  hasError,
  nextVisible,
  printVisible,
  previousVisible,
}) {
  return (
    <>
      <StyledDiv>
        {printVisible && (
          <StyledButton type="button" onClick={printResult}>
            Imprimer le résultat
          </StyledButton>
        )}
        {nextVisible && !previousVisible && (
          <StyledButton disabled={hasError} variant="primary">
            Commencer
            <ArrowIcon />
          </StyledButton>
        )}
        {nextVisible && previousVisible && (
          <StyledButton disabled={hasError} variant="primary">
            Suivant
            <ArrowIcon />
          </StyledButton>
        )}
        {previousVisible && (
          <StyledButton small type="button" onClick={onPrev} variant="flat">
            Précédent
          </StyledButton>
        )}
      </StyledDiv>
    </>
  );
}

PrevNextBar.propTypes = {
  hasError: PropTypes.bool,
  nextVisible: PropTypes.bool,
  onPrev: PropTypes.func,
  previousVisible: PropTypes.bool,
};
PrevNextBar.defaultProps = {
  hasError: false,
  nextVisible: true,
  onPrev: () => {},
  previousVisible: true,
};

export { PrevNextBar };

const { breakpoints, spacings } = theme;

const StyledButton = styled(Button)`
  & + & {
    @media (max-width: ${breakpoints.tablet}) {
      margin-top: ${spacings.base};
    }
  }
`;

const StyledDiv = styled.div`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: space-between;
  margin: 5rem 0 ${spacings.large} 0;
  @media (max-width: ${breakpoints.tablet}) {
    flex-flow: column;
    align-items: stretch;
  }
  @media print {
    display: none;
  }
`;

const ArrowIcon = styled(icons.DirectionRight)`
  width: 3rem;
  margin-left: ${spacings.small};
`;
