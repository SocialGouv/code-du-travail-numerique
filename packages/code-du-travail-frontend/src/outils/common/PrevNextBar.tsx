import { Button, icons, theme } from "@socialgouv/cdtn-ui";
import React, { FunctionComponent } from "react";
import styled from "styled-components";

import printResult from "./printResult";

type Props = {
  onPrev: () => void;
  hasError: boolean;
  nextVisible: boolean;
  printVisible: () => void;
  previousVisible: boolean;
  onNext: () => void;
};

export const PrevNextBar: FunctionComponent<Props> = ({
  onPrev,
  hasError = false,
  nextVisible = true,
  printVisible,
  previousVisible = true,
  onNext,
}) => {
  return (
    <>
      <StyledDiv>
        {printVisible && (
          <StyledButton type="button" onClick={printResult}>
            Imprimer le résultat
          </StyledButton>
        )}
        {nextVisible && !previousVisible && (
          <StyledButton disabled={hasError} variant="primary">
            Commencer
            <ArrowIcon />
          </StyledButton>
        )}
        {nextVisible && previousVisible && (
          <StyledButton disabled={hasError} variant="primary" onClick={onNext}>
            Suivant
            <ArrowIcon />
          </StyledButton>
        )}
        {previousVisible && (
          <StyledButton small type="button" onClick={onPrev} variant="flat">
            Précédent
          </StyledButton>
        )}
      </StyledDiv>
    </>
  );
};

const { breakpoints, spacings } = theme;

const StyledButton = styled(Button)`
  & + & {
    @media (max-width: ${breakpoints.tablet}) {
      margin-top: ${spacings.base};
    }
  }
`;

const StyledDiv = styled.div`
  display: flex;
  flex-direction: row-reverse;
  align-items: center;
  justify-content: space-between;
  margin: 5rem 0 ${spacings.large} 0;
  @media (max-width: ${breakpoints.tablet}) {
    flex-flow: column;
    align-items: stretch;
  }
  @media print {
    display: none;
  }
`;

const ArrowIcon = styled(icons.DirectionRight)`
  width: 3rem;
  margin-left: ${spacings.small};
`;
