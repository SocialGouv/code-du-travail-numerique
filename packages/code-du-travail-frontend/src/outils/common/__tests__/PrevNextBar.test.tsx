import { render } from "@testing-library/react";
import React from "react";

import { MatomoBaseEvent, MatomoSimulatorEvent } from "../../../lib";
import { matopush } from "../../../piwik";
import { PrevNextBar } from "../PrevNextBar";
import { printResult } from "../utils";

jest.mock("../../../piwik.js", () => ({
  matopush: jest.fn(),
}));

jest.spyOn(window, "print").mockImplementation();

describe("<PrevNextBar />", () => {
  it("should render", () => {
    const { container } = render(
      <PrevNextBar
        onPrev={jest.fn()}
        printVisible={false}
        hasError={false}
        nextVisible={false}
        previousVisible={false}
      />
    );

    expect(container).toMatchSnapshot();
  });
  it("should render only prev button", () => {
    const { container } = render(
      <PrevNextBar
        onPrev={jest.fn()}
        printVisible={false}
        hasError={false}
        nextVisible={false}
        previousVisible={true}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render only next button", () => {
    const { container } = render(
      <PrevNextBar
        onPrev={jest.fn()}
        printVisible={false}
        hasError={false}
        nextVisible={true}
        previousVisible={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render a disabled button", () => {
    const { container } = render(
      <PrevNextBar
        onPrev={jest.fn()}
        printVisible={false}
        hasError={false}
        nextVisible={false}
        previousVisible={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should trigger onPrev", () => {
    const onPrev = jest.fn();
    const { getByText } = render(
      <PrevNextBar
        onPrev={onPrev}
        printVisible={false}
        hasError={false}
        nextVisible={false}
        previousVisible={true}
      />
    );
    const prevButton = getByText(/précédent/i);
    prevButton.click();
    expect(onPrev).toHaveBeenCalled();
  });
  it("should trigger submit", () => {
    const onSubmit = jest.fn((event) => event.preventDefault());
    const { getByText } = render(
      <form onSubmit={onSubmit}>
        <PrevNextBar
          onPrev={jest.fn()}
          printVisible={false}
          hasError={false}
          nextVisible={true}
          previousVisible={true}
        />
      </form>
    );
    const nextButton = getByText(/suivant/i);
    nextButton.click();
    expect(onSubmit).toHaveBeenCalled();
  });

  it("should trigger a matomo event", () => {
    const simulatorName = "Test";
    const { getByText } = render(
      <PrevNextBar
        onPrev={jest.fn()}
        printVisible={true}
        hasError={false}
        nextVisible={false}
        previousVisible={false}
        onPrint={() => {
          printResult(simulatorName);
        }}
      />
    );
    const printButton = getByText(/imprimer/i);
    printButton.click();
    expect(matopush).toHaveBeenCalledWith([
      MatomoBaseEvent.TRACK_EVENT,
      MatomoBaseEvent.OUTIL,
      MatomoSimulatorEvent.CLICK_PRINT,
      simulatorName,
    ]);
    expect(window.print).toHaveBeenCalled();
  });
});
