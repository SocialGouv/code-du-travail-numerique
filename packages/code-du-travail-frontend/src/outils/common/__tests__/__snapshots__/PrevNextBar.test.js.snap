// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrevNextBar /> should render 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
  will-change: transform;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5rem 0 3.2rem 0;
}

.c3 {
  width: 3rem;
  margin-left: 1rem;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media print {
  .c4 {
    display: none;
  }
}

@media (max-width:980px) {
  .c2 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 "
    >
      Suivant
      <svg
        aria-hidden="true"
        class="c3"
        fill="none"
        viewBox="0 0 28 15"
      >
        <path
          d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="c4 c2 "
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render a disabled button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
  will-change: transform;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5rem 0 3.2rem 0;
}

.c3 {
  width: 3rem;
  margin-left: 1rem;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media print {
  .c4 {
    display: none;
  }
}

@media (max-width:980px) {
  .c2 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 "
      disabled=""
    >
      Suivant
      <svg
        aria-hidden="true"
        class="c3"
        fill="none"
        viewBox="0 0 28 15"
      >
        <path
          d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="c4 c2 "
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render only next button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 5.2rem;
  padding: 0 4.4rem;
  color: #fff;
  background: #f66663;
  border-color: #f66663;
  box-shadow: 0px 10px 20px rgba(246,102,99,0.26),0px 5px 7px rgba(246,102,99,0.35);
  opacity: 1;
  will-change: transform;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #f99593;
  border-color: #f99593;
}

.c1[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c4:link,
.c4:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c4:not([disabled]):hover,
.c4:not([disabled]):active,
.c4:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c4[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5rem 0 3.2rem 0;
}

.c3 {
  width: 3rem;
  margin-left: 1rem;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:600px) {
  .c1 {
    padding: 0 3rem;
  }
}

@media (max-width:600px) {
  .c4 {
    font-size: 1.4rem;
  }
}

@media print {
  .c4 {
    display: none;
  }
}

@media (max-width:980px) {
  .c2 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 "
    >
      Suivant
      <svg
        aria-hidden="true"
        class="c3"
        fill="none"
        viewBox="0 0 28 15"
      >
        <path
          d="M27.875 7.153l-6-6.284a.526.526 0 00-.711-.023.496.496 0 00-.024.688l5.204 5.468H.5c-.276 0-.5.217-.5.484s.224.483.5.483h25.844l-5.204 5.469a.493.493 0 00.024.687.52.52 0 00.711-.023l6-6.284a.506.506 0 000-.665z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="c4 c2 "
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;

exports[`<PrevNextBar /> should render only prev button 1`] = `
.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-sizing: content-box;
  font-weight: 500;
  font-size: 1.6rem;
  line-height: 1.125em;
  text-align: center;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 0.6rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,-webkit-transform 100ms linear;
  -webkit-transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  transition: background-color 250ms linear, border-color 250ms linear,transform 100ms linear;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 4rem;
  padding: 0 3rem;
  color: #3e486e;
  background: #fff;
  border-color: #bbcadf;
  box-shadow: none;
  opacity: 1;
  will-change: transform;
}

.c1:link,
.c1:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #3e486e;
}

.c1:not([disabled]):hover,
.c1:not([disabled]):active,
.c1:not([disabled]):focus {
  opacity: 1;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
  background: #fff;
  border-color: #dee5ef;
}

.c1[disabled] {
  background-color: #e4e8ef;
  border-color: #e4e8ef;
  color: #9298af;
  box-shadow: none;
  cursor: not-allowed;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 5rem 0 3.2rem 0;
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.4rem;
  }
}

@media print {
  .c1 {
    display: none;
  }
}

@media (max-width:980px) {
  .c2 + .c2 {
    margin-top: 1.6rem;
  }
}

@media (max-width:980px) {
  .c0 {
    -webkit-flex-flow: column;
    -ms-flex-flow: column;
    flex-flow: column;
    -webkit-align-items: stretch;
    -webkit-box-align: stretch;
    -ms-flex-align: stretch;
    align-items: stretch;
  }
}

@media print {
  .c0 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1 c2 "
      type="button"
    >
      Précédent
    </button>
  </div>
</div>
`;
